### PR TITLE
Feature/issue 173

### PR DESCRIPTION
**Pull request by mngweb**

### CODE REVIEW

File: backend/src/routes/api/users.routes.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  async function deleteUser(req, res, next) {
+    try {
+      await Profile.find({ user: res.user._id }).deleteOne();
+      await res.user.deleteOne();
+      res.status(StatusCodes.OK).json({ message: 'User deleted' });
+    } catch (err) {
+      next(err);
+    }
+  }
```
Comment (Nefariusek) [inline backend/src/routes/api/users.routes.js:32]: please, move those functions to userMiddlewares.js as well :D

File: backend/src/routes/api/users.routes.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { StatusCodes } from 'http-status-codes';
+import User, { userValidator } from '../../models/User.js';
+import Profile from '../../models/Profile.js';
+import getUserById from '../../middlewares/userMiddlewares.js';
+
+const UserRoutes = (router) => {
+  router.get('/users/:id', getUserById, async (req, res) => {
+    res.status(StatusCodes.OK).json(res.user);
+  });
```
Comment (Nefariusek) [inline backend/src/routes/api/users.routes.js:9]: even though this function is almost oneliner, please extract it to the middleware (maybe in the future we will make one middleware responsible for sending responses)

**Review by Nefariusek — CHANGES_REQUESTED** [review]

File: backend/src/routes/api/users.routes.js (hunk 1 above)
Comment (mngweb) [reply backend/src/routes/api/users.routes.js]: done

File: backend/src/routes/api/users.routes.js (hunk 2 above)
Comment (mngweb) [reply backend/src/routes/api/users.routes.js]: done :) I thought about it but missed it.

**Review by Nefariusek — APPROVED** [review]

**Pull request merged**